core/metadata-3.2.md

```diff
@@ -20,26 +40,139 @@ This proposal aims to codify one mechanism for working with metadata: metadata
 may be configured through a client-to-server event created specifically for
 this purpose.
 
-## METADATA
+## Mechanisms
+
+This document defines the following new protocol features:
+
+* Capability: `metadata`
+* Server notification: `METADATA`
+* Client command: `METADATA`
+* Server reply and error numerics
+
+### Capability
+
+The `metadata` capability and value indicates that a server supports metadata, and communicates any configuration value keys that may affect the level of support.
+
+The ABNF format of the `metadata` capability and value is: (TODO verify ABNF)
+
+    capability ::= 'metadata' ['=' value]
+    value      ::= value_item [',' value_item]*
+    value_item ::= ( 'maxsub=' integer ) |  ( 'maxkey=' integer )
```

Comment:
If we're intending this to be extensible we should probably do something more along the lines of:

```
    capability ::= 'metadata' ['=' tokens]
    tokens     ::= token [',' token]*
    token      ::= key ['=' value]
    key        ::= <sequence of a-zA-Z0-9.-_ >
    value      ::= // whatever
```
or whatevs. and then just below define the keys and their allowed values. for the ISUPPORT key we can just say use the same `tokens` format, `METADATA ['=' tokens]` etc.

Reply:
Extensibility sounds nice. I just hope clients get really aware of those and implement them.  Any strategy plan to avoid fragmentation and incompatibility?